test/test_reductions.py

```diff
@@ -1771,12 +1771,20 @@ def test_nansum_out_dtype(self, device, dtype):
         out_dtype = dtype
         inp_dtypes = all_types_and(torch.half) if out_dtype.is_floating_point else integral_types()
         for inp_dtype in inp_dtypes:
+            # TODO: Investigate why the output is not close to numpy.
```

Comment:
Is this flaky this way on all devices?

Also wouldn't it be less prone to floating point rounding error to compare to numpy in fp64 and then compared fp32 to fp64 ?

Reply:
Yes, it's flaky on all devices.  Observed by ROCm internal QA testing. But same problematic random inputs breaks the test for CUDA as verified on V100.  It's random which set of inputs you end up with.  I haven't observed CUDA failures in OSS CI for ROCm or CUDA.  It was our internal QA team that found the problematic random inputs.  I saved the fp32 inputs that pass and fail as npy files in a zip file to this PR.

Reply:
Sounds good.
Should we just re-use the values from sum() here: https://github.com/pytorch/pytorch/blob/e29004615f7ff59e75830c8c88c71a17001c3251/test/test_reductions.py#L1733-L1743 ? They should be consistent I guess

Reply:
(re-use as in sharing the values to ensure they stay consistent)

Reply:
Done, good idea.

Reply:
Sorry I wasn't clear. I meant, lift the sum values into a global enough scope that we can have these values defined once and re-used by both :D

Reply:
lol fixed again.  I should have applied the DRY principle in the first place.